nixos/modules/installer/tools/nixos-rebuild.sh

```diff
@@ -225,16 +245,47 @@ if [ -z "$_NIXOS_REBUILD_REEXEC" ]; then
     export PATH=@nix@/bin:$PATH
 fi
 
+# Use /etc/nixos/flake.nix if it exists. It can be a symlink to the
+# actual flake.
+if [[ -z $flake && -e /etc/nixos/flake.nix ]]; then
+    flake="$(dirname "$(readlink -f /etc/nixos/flake.nix)")"
+fi
+
 # Re-execute nixos-rebuild from the Nixpkgs tree.
-if [ -z "$_NIXOS_REBUILD_REEXEC" -a -n "$canRun" -a -z "$fast" ]; then
+# FIXME: get nixos-rebuild from $flake.
```

Comment:
This is important for [my workflow](https://xkcd.com/1172/)

Reply:
The re-exec is fundamentally broken anyway, since it tries to build `nixos-rebuild` with the old Nix. (The code to upgrade Nix happens later.) We could do `nix build $flake#nixosConfigurations.$flakeAttr.config.system.build.nixos-rebuild` but I don't think it's useful currently.